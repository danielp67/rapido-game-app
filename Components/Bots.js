import React from 'react';
import Player from "./Player";

const Bots = (props) => {

    const {droppedCard, setDroppedCard, gameStop, setScore, stop, pause, start, level, loading, settings, gamePause} = props
    const color = ["primary", "success", "danger", "warning", "primary", "success", "danger", "warning", "primary", "success", "danger"]

    return (
        <>
            {color.slice(0,settings.nbPlayer-1).map((color, index) => {


                    return (
                        <Player
                            key={index}
                            realPlayer={false}
                            playerIndex={index + 2}
                            color={color}
                            droppedCard={droppedCard}
                            setDroppedCard={setDroppedCard}
                            gameStop={gameStop}
                            start={start}
                            pause={pause}
                            stop={stop}
                            setScore={setScore}
                            level={level}
                            loading={loading}
                            gamePause={gamePause}
                        />

                    )
                }
            )}
        </>)

};


export default Bots;