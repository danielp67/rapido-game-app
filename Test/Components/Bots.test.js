import React from 'react';
import {create, act} from 'react-test-renderer';
import Bots from '../../Components/Bots';


test('card', () => {
    const deck = [{card: 't', suit: 't'}];
    const mockFn = jest.fn()
    const tree = create(<Bots
    start={true}
    stop={false}
    pause={false}
    droppedCard={{card: 't', suit: 't'}}
    setDroppedCard={mockFn}
    gameStop={false}
    setScore={mockFn}
    level={1000}
    loading={false}
    settings={deck}
    gamePause={false}
/>)
    expect(tree.toJSON()).toMatchSnapshot();
})