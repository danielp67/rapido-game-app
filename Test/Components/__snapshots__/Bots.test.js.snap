// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card 1`] = `<View />`;
